### ISSUE TITLE
Alternator can't restart when there are old-style (ka) sstables

### ISSUE THREAD

**Issue by nyh** [scylladb/scylla#6071]
A user reported the following errors (user-specific details redacted) while restarting Scylla with Alternator:
```
ERROR <date> [shard 0] database - Exception while populating keyspace 'alternator' with column family 'table-name:IndexName' from file '/var/lib/scylla/data/alternator/table-name:IndexName-<UUID>': sstables::malformed_sstable_exception (invalid version for file alternator-table-name:IndexName-ka-12-Data.db. Name doesn't match any known version.)
ERROR <date> [shard 0] database - Exception while populating keyspace 'alternator' with column family 'table-name' from file '/var/lib/scylla/data/alternator/table-name-<UUID>': sstables::malformed_sstable_exception (invalid version for file alternator-table-name-ka-13-Data.db. Name doesn't match any known version.
```

I don't know yet how to reproduce this issue - the presence of an "alternator" keyspace indicates that this is a relatively old sstable, because we now use per-table keyspaces and not a single "alternator" keyspace, and, moreover, I don't know why any "ka" tables were created in the first place.

But the direct cause for this error is clear: in commit 2fb2eb27a2cdfe9668d138ae46e9e71f6fab3df2 we fixed the regular expression for the newer sstable directory names (solving issue #5280), but we also need to fix the `ka` regular expression in `entry_descriptor::make_descriptor` to not only allow "`\w`" in table names.

Interestingly, in issue #5280 the problem we had in mind were the special characters like ":" we add in index names, but the second error message here shows we have the same problem when the table name contains a dash (in this case) or a dot - which are legal in DynamoDB table names - but not accepted by "`\w`".

**Comment by nyh** [comment]
@haaawk explained on the mailing list that:
> Nodes use KA/LA before they come to the agreement that all nodes in the cluster support MC. Only after this is agreed, each node starts to use MC.

@tgrabiec explained that the choice between KA and LA...
> is the same story as with MC. LA is enabled by a node only when it discovers that all nodes in the cluster support the LA_SSTABLE feature.
> Since 5dc3f5ea33f2c5a0a1a4c69e5a7a1fb5dc10c113 (>= 3.1.0), we persist sstable format in system.local, so it should persist across restarts. We also make sure it changes monotonically, once we start generating sstables in a given version that node should never revert back to generating sstables in older formats.

**Comment by nyh** [comment]
The antique KA format differed from the LA format in that instead of keyspaces being a directory, they were part of the sstable file name. A KA-format file name looks like: `keyspacename-tablename-ka-12-Data.db`. But this separator, "`-`", is unusable for Alternator because both keyspace name and table name may contain a dash, which is a legal character name in DynamoDB tables.

I think the proper solution is to stop writing new KA tables so we never encounter this problem. The only reason we write old sstables (temporarily, in a partially-booted cluster) is to allow downgrading a mixed cluster that failed an upgrade. But we left KA several **years** ago, nobody will want to downgrade back to KA, so there is no reason to write new KA files.

**Comment by nyh** [comment]
Sent a patch titled "sstable: default to LA format instead of KA format" to the mailing list.

Note that this patch will not "fix" sstables already written in the KA format and causing this problem. It will only prevent this problem from ever occurring again in the future. Because Alternator is not yet GA and we don't support upgrades yet, this is fine.

**Comment by nyh** [comment]
Already backported the important part of the fix (the change of default) to 4.0. No need to backport to earlier versions, as Alternator was experimental.
